packages/backend/src/tools/updateDiffHistory.ts

```diff
@@ -36,14 +37,42 @@ async function updateDiffHistoryFile() {
     getFileVersionOnMainBranch(`${discoveryFolder}/discovered.json`)
   const discoveryFromMainBranch =
     DiscoveryJsonFromMainBranch === ''
-      ? { contracts: [] }
+      ? undefined
       : (JSON.parse(DiscoveryJsonFromMainBranch) as DiscoveryOutput)
   const diff = diffDiscovery(
-    discoveryFromMainBranch.contracts,
+    discoveryFromMainBranch?.contracts ?? [],
     curDiscovery.contracts,
     config,
   )
 
+  // To check for changes to source code,
+  // download sources for block number from main branch
+  let codeDiff = ''
+  if (discoveryFromMainBranch) {
```

Comment:
What if the `blockNumberFromMainBranch` is the same as the already downloaded one? Should we check if it already exists or always remove and download? One is faster the other is more robust. Discovery is already slow as it is.

Reply:
For simplicity, I'll keep current solution. We can refactor when we decide that this functionality is useful.